### DISCUSSION TITLE
Is Linux offset problem was fixed?

### DISCUSSION THREAD

**Discussion by okolenmi** [ppy/osu#30949] [Q&A]
1 month ago it was ok to use +4ms offset on Linux to have keyboard sounds + Music fit perfectly in rhythm but now I have to use offset near -12 to get the same results. I have 2 theories:
- **audio performance become worse**  - unlikely, as I have a good gameplay feeling)
- **15 ms offset that intended to be only for Windows was implemented on Linux as well, and now it's removed**  - strange thing as I see this task marked as completed a long time ago

**Comment by peppy** [comment]
To my knowledge we changed nothing, so it might be a change in the alignment of LinuxStarsAndPlanets

**Reply by okolenmi** [reply]
![2024-12-03 092252](https://github.com/user-attachments/assets/0b11306f-6695-4073-b725-09d81db6eb79)

~30ms... Something has been changed in the game or Linux as my setup and settings were the same all the time. Also it means 15ms hidden offset is still in game on Linux for some reason. I will try to find out a reason of extra ~15ms.

**Reply by okolenmi** [reply]
It looks like it's not a problem of Linux - version was reverted to 6.8.0-45 (after hours of pain with manual driver updates) and there is still 30ms latency instead of old ~15 or less.

**Reply by okolenmi** [reply]
Low latency kernel, low latency JACK audio configuration (44.1K 128 = 2.9ms) and still everything I got is better audio quality by price of this 2.9ms... It doesn't look like problem is in OS. It's not a problem of input as well - the game feels good except late hitsounds. Input latency from mouse and keyboard the same. I have recorded gameplay without music (hitsounds oly + my keyboard/mose)  and analyzed it a lot of times. I have no progress after all my attempts.
![2024-12-03 205811](https://github.com/user-attachments/assets/99d9746d-a30d-4995-a0fe-069fe1bb78de)
